### PR TITLE
Fpo 294 storing email data

### PR DESCRIPTION
**Pull request by mkayad**
### Jira link
[FPO-294](https://transformuk.atlassian.net/browse/FPO-294)
### What?

I have added/removed/altered:

- [x] Added store user data to database
- [x] Removed
- [x] Altered

### Why?

I am doing this because:

- it is needed to verify the user application

### AC

- Tested
- Shared with stakeholders
- Pair reviewed

### CODE REVIEW

File: .bin/seed.sh [hunk 1]
```diff
@@ -123,5 +123,5 @@ create_user "1234" "local-development" "Unregistered"
 create_user "2345" "local-development" "Unregistered"
 create_user "3456" "local-development" "Unregistered"
 create_user "4567" "local-development" "Unregistered"
```
Comment (ashward) [inline .bin/seed.sh:125]: Won't these create users with an email address of `Unregistered`?

File: .bin/seed.sh [hunk 2]
```diff
@@ -104,7 +104,7 @@ create_user() {
         --table-name "$USERS_TABLE_NAME" --item "{
             \"UserId\": {\"S\": \"$id\"},
             \"OrganisationId\": {\"S\": \"$organisationId\"},
-            \"Status\": {\"S\": \"$status\"},
```
Comment (ashward) [inline .bin/seed.sh:107]: Do we not still want to be able to set the `Status` of the test fixture?

File: src/controllers/usersController.ts [hunk 3]
```diff
@@ -40,12 +40,33 @@ export class UserController {
...
     res.status(201).json({ organisationId })
   }
+
+  async getOrganisation (req: Request, res: Response): Promise<void> {
+    const organisationId = req.params.organisationId
+    const organisation = await this.organisationRepository.getOrganisation(organisationId)
+    if (organisation === null) {
+      res.status(404).json({ message: 'organisation not found' })
+    } else {
+      res.status(201).json({ ...organisation.toJson() })
```
Comment (ashward) [inline src/controllers/usersController.ts:69]: Why is this a `201` response? Surely it should be a `200`?

File: .bin/seed.sh (hunk 2 above)
Comment (mkayad) [reply .bin/seed.sh:107]: that is correct, Status is not needed in Users table anymore

File: .bin/seed.sh (hunk 1 above)
Comment (mkayad) [reply .bin/seed.sh:125]: correct

File: src/models/utils.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export default function getFieldValue (plainObject: any, field: string): string {
```
Comment (willfish) [inline src/models/utils.ts:1]:
```suggestion
export default function getNestFieldValue (plainObject: any, field: string): string {
```

File: spec/operations/updateUserSpec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import 'jasmine'
+import { type DynamoDBClient } from '@aws-sdk/client-dynamodb'
+import { UpdateUser } from '../../src/operations/updateUser'
+
+describe('CreateUser', () => {
```
Comment (willfish) [inline spec/operations/updateUserSpec.ts:5]:
```suggestion
describe('UpdateUser', () => {
```

File: spec/controllers/userControllerSpec.ts [hunk 6]
```diff
@@ -119,13 +121,61 @@ describe('UserController', () => {
       await controller.create(req, res)
 
       expect(userRepository.createUser).toHaveBeenCalledWith('id', 'organisationId')
-      expect(organisationRepository.getOrganisation).toHaveBeenCalledWith('organisationId')
```
Comment (willfish) [inline spec/controllers/userControllerSpec.ts:122]: So I don't think this behaviour of implicitly finding/creating the organisation has changed?

File: src/controllers/usersController.ts (hunk 3 above)
Comment (willfish) [inline src/controllers/usersController.ts:69]: Should this be 200?

File: .bin/seed.sh (hunk 2 above)
Comment (willfish) [reply .bin/seed.sh:107]: So probably for a separate PR we'll want to add an ability to seed the organisation for a given user and tie them together

File: src/controllers/usersController.ts (hunk 3 above)
Comment (mkayad) [reply src/controllers/usersController.ts:69]: agreed

Comment (mkayad) [reply src/controllers/usersController.ts:69]: agreed, we are not created a new resource, just updating an existing one.

**Comment by willfish** [comment]
LGTM. A small question about the lost test around implicit finding/creating organisations/some typos but basically looks there.

We're in a bit of a spaghetti state doing the find/create of the organisation in the show action. I also find it weird having organisation routes pointing at the users controller. Normally you decompose these concepts and avoid getting into confusing states. My instincts are that there's not too much more data model changes to come and its not like this app is going to be a mainstay of auth for other protected resources in the future so its probably fine to just preserve this implementation as is.

Coming back to this I'm a little bit blown away by how much boiler plate there is in the tests just to do basic mocking and such

File: spec/operations/updateUserSpec.ts (hunk 5 above)
Comment (mkayad) [reply spec/operations/updateUserSpec.ts:5]: correct

File: spec/controllers/userControllerSpec.ts (hunk 6 above)
Comment (mkayad) [reply spec/controllers/userControllerSpec.ts:122]: No, it did not change

**Pull request merged**